### ISSUE TITLE
When adding task prior to currrent time, the date gets all confused

### ISSUE THREAD

**Issue by scottdky** [projecthamster/hamster#646]
When adding tasks previous to the current time, the date gets confused and ends up messing up previous entries if the time overlaps the new task time period. This bug applies to  v3.0.2. 

To reproduce, add a new task and use the calendar tool to set the date to a prior day in the week. In the cmdline field, the date will be one day prior to the one chosen in the calendar. If this incorrect date is not noticed, then previous activities are compromised. There are also other problems when choosing dates. The following should be changed or fixed:

1. The cmdline date should match that chosen in the calendar tool.
2. The date shown below the start and end fields should not change automatically but only when set directly by the user. The only exception is when the start date is chosen, the end date can change to be the same day.
3. Adding a new task should not alter existing tasks. If there is a conflict when attempting to save, an alert should come up to notify the user of the conflict. Dismissing the alert simply returns the user to the activity window.
4. The save button should not be disabled due to an activity conflict. Item 3 should be done instead. As an alternative to item 3, the conflicting fields could be highlighted along with an error message below the fields showing the problem similar to what is done in web forms. In this scenario, disabling the save button would be perfectly acceptable.
5. The save button is disabled for unknown reasons. Conflicting tasks can be entered and saved without issue. Other times the save button is disabled. I do not know if it was due to a conflict or not.

**Comment by GeraldJansen** [comment]
The misalignment between the date selected in the calendar and that inserted in the command line is a known bug with a good fix  awaiting review since August 14 (PR #630). Reviewing and/or testing that PR would probably accelerate its acceptance.

A review of the handling of overlapping activities is an even longer standing but separate issue (eg. #396).

Aside from the absence of any activity, the main reason for the Save button being disabled is that the end date-time is less than the start date-time. In this case, if you hover over the Save button a popup will appear with a message about a negative duration.  I agree that the user experience is sub-optimal in this respect.

**Comment by matthijskooijman** [comment]
I indeed believe that 1 and 2 are essentially duplicate of #590 and will be fixed once #630 is merged.

As for 3, as @GeraldJansen points out, that is related to #396. Your suggestion of asking for confirmation before modifiying existing activities makes sense to me, maybe you could add that suggest to #396 so we can keep that discussion in one place?

As for 4 and 5, improving the feedback when the save button is disabled seems like a good and specific suggestion. With the other points being addressed elsewhere, I'm taking the liberty to repurpose this issue to be about just that suggestion. If that doesn't seem right, let me know and we can reconsider.

**Comment by ederag** [comment]
> good fix awaiting review since August 14 (PR #630)

A better fix has been available since May 1 (more than three months earlier): https://github.com/projecthamster/hamster/pull/597#issuecomment-622463737.
